pvlib/iam.py

```diff
@@ -748,3 +750,267 @@ def marion_integrate(function, surface_tilt, region, num=None):
+       rendering. Computer graphics forum 13 (1994).
+
+    .. [2] Xie, Y., M. Sengupta, A. Habte, A. Andreas, "The 'Fresnel Equations'
+       for Diffuse radiation on Inclined photovoltaic Surfaces (FEDIS)",
+       Renewable and Sustainable Energy Reviews, vol. 161, 112362. June 2022.
+       :doi:`10.1016/j.rser.2022.112362`
+
+    See Also
+    --------
+    pvlib.iam.schlick
+    pvlib.iam.fedis_diffuse
+    """
+    # these calculations are as in [2]_, but with the refractive index
+    # weighting coefficient w set to 1.0 (so it is omitted)
+
+    # relative transmittance of sky diffuse radiation by PV cover:
+    cosB = cosd(surface_tilt)
+    sinB = sind(surface_tilt)
+    cuk = (2 / (np.pi * (1 + cosB))) * (
+        (30/7)*np.pi - (160/21)*np.radians(surface_tilt) - (10/3)*np.pi*cosB
+        + (160/21)*cosB*sinB - (5/3)*np.pi*cosB*sinB**2 + (20/7)*cosB*sinB**3
+        - (5/16)*np.pi*cosB*sinB**4 + (16/105)*cosB*sinB**5
+    )  # Eq 4 in [2]
+
+    # relative transmittance of ground-reflected radiation by PV cover:
+    with np.errstate(divide='ignore', invalid='ignore'):  # Eq 6 in [2]
+        cug = 40 / (21 * (1 - cosB)) - (1 + cosB) / (1 - cosB) * cuk
+
+    cug = np.where(surface_tilt < 1e-6, 0, cug)
+
+    # respect input types:
+    if np.isscalar(surface_tilt):
+        cuk = cuk.item()
+        cug = cug.item()
+    elif isinstance(surface_tilt, pd.Series):
+        cuk = pd.Series(cuk, surface_tilt.index)
+        cug = pd.Series(cug, surface_tilt.index)
+
+    return cuk, cug
+
+
+def fedis(aoi, n=1.5, n_ref=None):
+    """
+    Determine the incidence angle modifier (IAM) for direct irradiance
+    using the FEDIS transmittance model.
+
+    Note that FEDIS [1]_ calculates direct IAM using the Fresnel equations
+    without extinction, thus in the default case of ``n_ref=None``, this is
+    equivalent to calling :py:func:`physical` with ``K=0``.
+
+    Parameters
+    ----------
+    aoi : numeric
+        Angle of incidence. [degrees]
+
+    n : float, default 1.5
+        Refractive index of the PV front surface material.  The default value
+        of 1.5 was used for an IMT reference cell in [1]_. [unitless]
+
+    n_ref : float, optional
+        Reference refractive index. If None (default), set equal to ``n``.
+        In [1]_ ``n_ref`` was set to 1.4585 for
+        a fused silica dome over a CMP22, but in conventional
+        PV applications it is appropriate to set ``n_ref=n`` (the default
+        behavior).
+
+    Returns
+    -------
+    iam : numeric
+        The incident angle modifier
+
+    References
+    ----------
+    .. [1] Xie, Y., M. Sengupta, A. Habte, A. Andreas, "The 'Fresnel Equations'
+       for Diffuse radiation on Inclined photovoltaic Surfaces (FEDIS)",
+       Renewable and Sustainable Energy Reviews, vol. 161, 112362. June 2022.
+       :doi:`10.1016/j.rser.2022.112362`
+
+    See Also
+    --------
+    pvlib.iam.physical
+    """
```

Comment:
```
def fedis(aoi, n=1.5, n_ref=None):

    iam_physical = physical(aoi, n, K=0, L=0)

    if n_ref is None:
        return iam_physical

    else:
        # reflectance for normal incidence with refractive index n:
        rd0 = ((n - 1) / (n + 1)) ** 2

        # reflectance for normal incidence with refractive index n_ref:
        r0 =  ((n_ref - 1) / (n_ref + 1)) ** 2

        # weighting function
        w = (1 - rd0) / (1 - r0)
        
        return w * iam_physical
```

Reply:
I wondered if something like this could be done but did not bother to look into it.  Thanks!